doc/guide/build_windows.hpp

```diff
@@ -0,0 +1,95 @@
+/*! @page build_windows Building mlpack From Source
+
+@section build_intro Introduction
+
+This document discusses how to build mlpack for Windows from source, so you can later
+create your own C++ applications.
+
+@section build_env Environment
+
+This tutorial has been designed and tested using:
+- Windows 10
+- Visual Studio 2017 (toolset v141)
+- mlpack-3.0.2
+- OpenBLAS.0.2.14.1
+- boost_1_66_0-msvc-14.1-64
+- armadillo-8.500.1
+- and x64 configuration
+
+The directories and paths used in this tutorial are just for reference purposes.
+
+@section build_prereqs Pre-requisites
+
+- Install CMake for Windows (win64-x64 version from https://cmake.org/download/)
+and make sure you can use it from the Command Prompt (may need to add to the PATH)
+
+- Download the latest mlpack release from here:
+<a href="http://www.mlpack.org/files/mlpack-3.0.2.tar.gz">mlpack-3.0.2</a>
+
+@section build_instructions Windows build instructions
+
+- Unzip mlpack to "C:\mlpack\mlpack-3.0.2"
```

Comment:
Is there any way to use an existing home directory? Something like ``~/``on linux.

Reply:
This is tricky - we could use %userprofile% in Windows, but that directory is not very friendly for the purpose of an easy-to-follow tutorial (and it changes depending on the windows version) - I think for simplicity we should use the most basic directory we can imagine (there is also a note that says this path is just for reference). Long paths or paths with spaces may produce issues in Windows so we are aiming for the safest option.

Let me know if you still want to change it.